### PR TITLE
ci: add arm64 runner

### PR DESCRIPTION
**Pull request by leseb**
Thanks to @xin3liang we now have 2 hosted runners on ARM64. So let's add
a small integration test for it.

Signed-off-by: Sébastien Han <seb@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

[skip ci]

### CODE REVIEW

**Comment by leseb** [comment]
The arm build kinda takes a while so perhaps we should run a nightly job for it. Now on all PRs.

File: .github/workflows/canary-integration-test-arm64.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    - name: print k8s cluster status
+      run: tests/scripts/github-action-helper.sh print_k8s_cluster_status
+
+    - name: use local disk and create partitions for osds
+      run: |
+        tests/scripts/github-action-helper.sh use_local_disk
+        tests/scripts/create-bluestore-partitions.sh --disk "$BLOCK" --osd-count 1
+
+    - name: build rook
+      run: tests/scripts/github-action-helper.sh build_rook
```
Comment (leseb) [inline .github/workflows/canary-integration-test-arm64.yml:53]: do not build, just fetch the image from docker hub master.

**Comment by xin3liang** [comment]
Do we need to change this line also, maybe the runner OS broken is relevant to this.
```
 sed -i "s|#deviceFilter:|deviceFilter: $(lsblk | awk '/14G/ {print $1}' | head -1)|g" cluster-test.yaml
```
https://github.com/rook/rook/blob/master/tests/scripts/github-action-helper.sh#L116

**Comment by xin3liang** [comment]
> Do we need to change this line also, maybe the runner OS broken is relevant to this.
> 
> ```
>  sed -i "s|#deviceFilter:|deviceFilter: $(lsblk | awk '/14G/ {print $1}' | head -1)|g" cluster-test.yaml
> ```
> 
> https://github.com/rook/rook/blob/master/tests/scripts/github-action-helper.sh#L116

How about 
```
sed -i "s|#deviceFilter:|deviceFilter: ${BLCOK}|g" cluster-test.yaml
```

File: tests/scripts/github-action-helper.sh [hunk 2]
```diff
@@ -19,7 +19,7 @@ set -xe
 #############
 # VARIABLES #
 #############
-BLOCK=$(sudo lsblk --paths | awk '/14G/ {print $1}' | head -1)
+: "${BLOCK:=$(sudo lsblk --paths | awk '/14G/ {print $1}' | head -1)}"
```
[the remark this answers is not in the document]
Comment (leseb) [reply tests/scripts/github-action-helper.sh:22]: This already what the line does.

[the remark this answers is not in the document]
Comment (xin3liang) [reply tests/scripts/github-action-helper.sh:22]: > This already what the line does.

Yeah, ":=" does it.

**Review by xin3liang — APPROVED** [review]

**Comment by leseb** [comment]
The job is passing CI now:

```
[root@rook-ceph-tools-78cdfd976c-k5bpc /]# ceph -s
  cluster:
    id:     af4f96ac-0ab7-4d22-ac36-770d3083bcb2
    health: HEALTH_WARN
            mon is allowing insecure global_id reclaim
            insufficient standby MDS daemons available

  services:
    mon:        1 daemons, quorum a (age 5m)
    mgr:        a(active, since 52s)
    mds:        1/1 daemons up
    osd:        1 osds: 1 up (since 3m), 1 in (since 4m)
    rbd-mirror: 1 daemon active (1 hosts)
    rgw:        1 daemon active (1 hosts, 1 zones)

  data:
    volumes: 1/1 healthy
    pools:   11 pools, 177 pgs
    objects: 243 objects, 13 KiB
    usage:   13 MiB used, 100 GiB / 100 GiB avail
    pgs:     177 active+clean

  io:
    client:   37 KiB/s rd, 1.1 KiB/s wr, 48 op/s rd, 28 op/s wr
```


MDSs are a bit flappy though.

File: .github/workflows/canary-integration-test-arm64.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    - name: teardown minikube and docker
+      run: |
+        uptime
+        minikube delete
+        docker system prune -a
+
+    - name: setup minikube
+      run: |
+        # sudo apt-get install build-essential -y
```
Comment (travisn) [inline .github/workflows/canary-integration-test-arm64.yml:29]: Shall we remove all these commands that are commented?

Comment (leseb) [reply .github/workflows/canary-integration-test-arm64.yml:29]: I decided to keep them if the runner goes away, so we have the entire procedure to set it up.

**Review by travisn — APPROVED** [review]

**Pull request merged**